x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Index pattern doesn\'t match any indices.');
+        });
+
+        it('should not allow an index pattern without time fields', async () => {
+          mockIndexPatternValidityResponse({ dateFields: [] });
+          await form.setInputValue('rollupIndexPattern', 'abc', true);
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Index pattern must match indices that contain time fields.');
+        });
+
+        it('should not allow an index pattern that matches a rollup index', async () => {
+          mockIndexPatternValidityResponse({ doesMatchRollupIndices: true });
+          await form.setInputValue('rollupIndexPattern', 'abc', true);
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Index pattern must not match rollup indices.');
+        });
+
+        it('should not be the same as the rollup index name', async () => {
+          await form.setInputValue('rollupIndexPattern', 'abc', true);
+          await form.setInputValue('rollupIndexName', 'abc', true);
+
+          userActions.clickNextStep();
+
+          const errorMessages = getFormErrorsMessages();
+          expect(errorMessages).toContain('Index pattern cannot have the same as the rollup index.');
+          expect(errorMessages).toContain('Rollup index cannot have the same as the index pattern.');
+        });
+      });
+
+      describe('rollup index name', () => {
+        beforeEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(false);
+        });
+
+        afterEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(true);
+        });
+
+        it('should not allow spaces', () => {
+          form.setInputValue('rollupIndexName', 'with space');
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Remove the spaces from your rollup index name.');
+        });
+
+        it('should not allow invalid characters', () => {
+          const expectInvalidChar = (char) => {
+            form.setInputValue('rollupIndexName', `rollup_index_${char}`);
+            userActions.clickNextStep();
+            expect(getFormErrorsMessages()).toContain(`Remove the characters ${char} from your rollup index name.`);
+          };
+
+          [...INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE, ','].reduce((promise, char) => {
+            return promise.then(() => expectInvalidChar(char));
+          }, Promise.resolve());
+        });
+
+        it('should not allow a dot as first character', () => {
+          form.setInputValue('rollupIndexName', '.kibana');
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Index names cannot begin with periods.');
+        });
+      });
+
+      describe('rollup cron', () => {
```

Comment:
Similarly, my instinct would be to move these tests into test file that's specific to the cron editor component, i.e. `steps/components/cron_editor/cron_editor.test.js`. Could you help me understand the rationale for putting them here?

Reply:
If we were to create the cron editor as a reusable component, it might make sense. But we don't so I prefer to test it in integration.

This article helps in understanding the reasoning: https://kentcdodds.com/blog/write-tests

> Integration tests strike a great balance on the trade-offs between confidence and speed/expense. This is why it's advisable to spend most (not all, mind you) of your effort there.